mlflow/models/resources.py

```diff
@@ -32,8 +30,9 @@ class Resource(ABC):
         target_uri (str): The target URI where these resources are hosted.
     """
 
-    type: ResourceType
-    target_uri: str
+    def __init__(self, type: ResourceType, target_uri: str):
```

Comment:
I'd make `type` an abstract property if that's required to ensure all the subclasses have it.

Reply:
agreed! :)

Reply:
I think `target_uri` can be made as abstract property too.

Reply:
@WeichenXu123 Why? Does each subclass have a different target URI?